feat: Add information about disabled edit/enable/create button in organization (GH-9991)

Parthlakhani opened the pull request:
closes #9991

Parthlakhani committed:
feat: Add information about disabled edit/enable button (GH-9991)

Parthlakhani committed:
refactor: change message style

Parthlakhani committed:
feat: Add shared component for edit/enable message (GH-9991)

Parthlakhani committed:
refactor: Change component name

Parthlakhani committed:
feat: Add explain disable message to other components

Parthlakhani committed:
refactor: Change explain disable messages

Parthlakhani committed:
feat: Add explain disable messages component in unit links

Pucek9 committed:
Merge branch 'develop' of https://github.com/SAP/cloud-commerce-spartacus-storefront into feature/GH-9991

Pucek9 committed:
let's confirm that it's ok

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani committed:
feat: Add condition for messages in link of units

Added support for disabled create button in explain disable message component.
Fixed CSS for single message.

Parthlakhani committed:
refactor: reset start script

Parthlakhani committed:
refactor:  Update logic for displaying disabled message

Pucek9 commented on feature-libs/organization/administration/components/shared/card/card.component.html:
```diff
@@ -29,6 +29,7 @@ <h4>
 
   <div class="main">
     <cx-org-message></cx-org-message>
+    <ng-content select="[messages]"></ng-content>
```
```suggestion
    <ng-content select="[info]"></ng-content>
```
let use some other tag

Pucek9 commented on feature-libs/organization/administration/components/shared/detail/explain-disable-messages/explain-disable-messages.component.ts:
```diff
@@ -0,0 +1,46 @@
+  /**
+   * Flag for disabledCreate message
+   */
+  @Input() disabledCreateMessage?: string;
```
why string? as you use it as boolean it's unexpected

Parthlakhani committed:
Update feature-libs/organization/administration/components/shared/card/card.component.html

Co-authored-by: Michał Gruca <pucek9@gmail.com>

Parthlakhani committed:
feat: make messages configurable (GH-9991)

closes #9991
Added logic for messages to display based on configuration.
Changed messages/main class to main.
Updated CSS sturcture

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
styling proposal

Pucek9 committed:
up

Parthlakhani committed:
feat: add custom message support (GH-9991)

Parthlakhani committed:
refactor: add comment for enableCustomMessage

Parthlakhani committed:
refactor: Update logic

Parthlakhani committed:
Add custom message support

Pucek9 committed:
cleanup

Pucek9 committed:
revert changes

Pucek9 committed:
add dots

Pucek9 committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
revert config in input name

Pucek9 committed:
Merge branch 'feature/GH-9991' of https://github.com/SAP/cloud-commerce-spartacus-storefront into feature/GH-9991

Pucek9 commented:
QA Steps: 
Verify custom messages displayed in cases:
- [x] 1 message for disabled purchase limit, 0 messages for purchase limit with disabled unit
- [x] 1 message for disabled budget, 2 messages for disabled budget with disabled unit
- [x] 1 message for disabled cc, 2 messages for disabled cc with disabled unit
- [x] 1 message for disabled user, 2 messages for disabled user with disabled unit
 unit
- [x] 1 message for disabled user, 2 messages for disabled user with disabled unit
- [x] 1 message for disabled unit, 2 messages for disabled unit with disabled parent unit, 1 message for root unit

Add custom content to the message, eg, cost center details component:
```ts
  <cx-org-explain-disable-messages info i18nRoot="orgCostCenter" [forceDisplay]="true">
    <li>Custom message</li>
  </cx-org-explain-disable-messages>
```  
Verify that Custom message is added to list and displayed

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani committed:
fix: fix lint error

Matejk00 commented:
Hey, I did QA for this ticket. All scenarios works fine. However I've found one styling issue. 
![css](https://user-images.githubusercontent.com/6958810/108194795-5178f480-7117-11eb-8810-835fb2bfb2c0.png)

In my opinion text in message box should be aligned with info icon. Right now text is positioned slighthly to the bottom.

@Parthlakhani what do you think?

Matejk00 reviewed (changes requested):
https://github.com/SAP/spartacus/pull/11008#issuecomment-780477645

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani commented:
> Hey, I did QA for this ticket. All scenarios works fine. However I've found one styling issue.
> ![css](https://user-images.githubusercontent.com/6958810/108194795-5178f480-7117-11eb-8810-835fb2bfb2c0.png)
> 
> In my opinion text in message box should be aligned with info icon. Right now text is positioned slighthly to the bottom.
> 
> @Parthlakhani what do you think?

I agree, I am checking how I can fix this.

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani committed:
style: fix icon alignment with list

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani committed:
test: Add unit tests (GH-9991)

Parthlakhani committed:
style: change padding and class

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
wip

Pucek9 committed:
Merge branch 'develop' of https://github.com/SAP/cloud-commerce-spartacus-storefront into feature/GH-9991

Parthlakhani committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
wip2

Pucek9 committed:
wip2

Pucek9 committed:
Merge branch 'feature/GH-9991' of https://github.com/SAP/cloud-commerce-spartacus-storefront into feature/GH-9991

Parthlakhani committed:
test: Update unit tests

Pucek9 committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
up

Parthlakhani committed:
test: Update unit tests

Parthlakhani committed:
test: update unit tests

Parthlakhani committed:
test: update unit tests

Pucek9 committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 commented on feature-libs/organization/administration/components/shared/detail/explain-disable-messages/explain-disable-messages.component.spec.ts:
```diff
@@ -0,0 +1,225 @@
+        component.ngOnInit();
+        current$.next(action.currentValue);
+        fixture.detectChanges();
+        const element = fixture.debugElement
```
its no longer element, its array of string now
```suggestion
        const values = fixture.debugElement
```
the same in other occurences

Pucek9 commented on feature-libs/organization/administration/components/shared/detail/explain-disable-messages/explain-disable-messages.component.spec.ts:
```diff
@@ -0,0 +1,225 @@
+        const element = fixture.debugElement
+          .queryAll(By.css('section > ul > li'))
+          .map((el) => el.nativeNode.innerText);
+        expect(element).not.toEqual(action.expectedValue);
```
instead of verify `.not.toEqual` we can use the same array and logic as above and provide an empty array as expected value 🙂

Parthlakhani committed:
Update feature-libs/organization/administration/components/shared/detail/explain-disable-messages/explain-disable-messages.component.spec.ts

Co-authored-by: Michał Gruca <pucek9@gmail.com>

Parthlakhani committed:
test: update unit tests

Parthlakhani committed:
refactor: change unit test

Pucek9 committed:
DDT

Parthlakhani committed:
fix: fix failing scenario for disableDisabled

Pucek9 committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
update

Pucek9 committed:
change 'messages' to 'info'

Pucek9 committed:
change 'messages' to 'info' in spec

Pucek9 committed:
up

Pucek9 committed:
up

Pucek9 committed:
up

Pucek9 committed:
changed order of information

Pucek9 committed:
add missing service to public api

Pucek9 committed:
re-arrange class

Pucek9 committed:
Merge branch 'develop' into feature/GH-9991

Pucek9 committed:
fix unit

Pucek9 committed:
Merge branch 'develop' into feature/GH-9991
